### PR TITLE
doc: Adding a compile from source instruction for (interested) beginners 

### PR DESCRIPTION
**Pull request by smota**

### CODE REVIEW

File: website/pages/docs/developers/build.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Build From Source
+
+The preferred way to use CloudQuery is through the available distribution, see the Quickstart section on the [Docs](https://www.cloudquery.io/docs)
+
+If you want to build it from the source, follow the steps:
+
+1. Cloudquery is developed in GO. Ensure you have a working [GO runtime](https://go.dev/)
```
Comment (candiduslynx) [inline website/pages/docs/developers/build.md:7]: Change `GO` to `Go`

File: website/pages/docs/developers/build.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+2. Clone or download the [code repository](https://github.com/cloudquery/cloudquery)
+3. On the source code root directory execute (it will recursively run GO in all subdirectories)
+>./scripts/build.sh
```
Comment (candiduslynx) [inline website/pages/docs/developers/build.md:10]:
````suggestion
```sh copy
./scripts/build.sh
```
````

File: website/pages/docs/developers/build.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+Cloudquery binaries will be created in a new folder *"bin"* in the root directory
```
Comment (candiduslynx) [inline website/pages/docs/developers/build.md:12]:
```suggestion
CloudQuery binaries will be created in a new `bin` directory
```

File: plugins/source/salesforce/client/client.go [hunk 4]
```diff
@@ -16,8 +16,6 @@ import (
 	"github.com/rs/zerolog"
 )
 
-const ApiVersion = "v56.0"
```
Comment (candiduslynx) [inline plugins/source/salesforce/client/client.go:19]: This change doesn't belong to this PR.
The API Version behavior change should be done in a separate PR, clearly explaining the need for this change as well as addressing the issue of the differences between the API responses for different versions.

**Review by candiduslynx — COMMENTED** [review]
Hi @smota!
Thanks for your contribution.

I think that the `saleforce` source plugin changes should be done in a separate PR with `feat(salesforce)!: NNN` name, so that it appears in the release notes.
Additionally, take a note of the `!` there, as this would be a breaking change and will require a major version bump.

As for the documentation changes: there are several comments there as well. Overall looks good.

**Review by candiduslynx — DISMISSED** [review]
Need to extract `salesforce` changes to a separate PR.

**Review by erezrokah — APPROVED** [review]
Thanks for the PR @smota, based on @candiduslynx I did a quick follow up so we can get this merged quickly 🚢 Feel free to add more comments on the PR and we can follow up
